### PR TITLE
SAMZA-2239 : Using jsonserde to write metadata file

### PR DESCRIPTION
**Pull request by rmatharu**

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java [hunk 1]
```diff
@@ -49,26 +50,34 @@
...
     if (metadataFile.isDefined()) {
-
-      StringBuilder metadata = new StringBuilder("Version: 1");
-      metadata.append(System.lineSeparator());
       MetricsHeader metricsHeader =
           new MetricsHeader(jobName, jobId, "samza-container-" + containerId, execEnvContainerId.orElse(""), LocalContainerRunner.class.getName(),
               Util.getTaskClassVersion(config), Util.getSamzaVersion(), Util.getLocalHost().getHostName(),
               System.currentTimeMillis(), System.currentTimeMillis());
 
+      class MetadataFileContents {
```
Comment (Sanil15) [inline samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java]: Can we put the inner class at the bottom?

Comment (rmatharu) [reply samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java]: no, this class is internal to the method not internal to the class.

**Comment by shanthoosh** [comment]
@rmatharu Looks like the travis-build is failing with compilation errors. Can you please fix it.

**Comment by mynameborat** [comment]
Can you please create a JIRA and attach it with the PR?

**Comment by shanthoosh** [comment]
@rmatharu 
Thanks for the patch. Merged to trunk.

**Pull request merged**